samples/graph-activity-feed-broadcast/nodejs/server.js

```diff
@@ -108,109 +103,91 @@ app.post('/SendNotificationToOrganisation', async (req, res) => {
-      let appList = await client.getInstalledAppsForUser(users.id);
-      if(appList){
-        let userAppId = getAppId(appList);
-        var encodedContext = encodeURI('{"subEntityId": ' + req.body.id + '}');
-          const postData = {
-            "topic": {
-              "source": "text",
-              "value": req.body.title,
-              "webUrl": 'https://teams.microsoft.com/l/entity/' + appId + '/broadcast?context=' + encodedContext
-            },
-            "activityType": "approvalRequired",
-            "previewText": {
-              "content": "Broadcast by"+ req.body.userName
-            },
-            "templateParameters": [
-              {
-                "name": "approvalTaskId",
-                "value": req.body.title
-              }
-            ]
-          };
-
+  var recipientsList = [];
+
+  if (userList.value) {
+    await forEachAsync(userList.value, async (users) => {
+      try {
+        let appList = await client.getInstalledAppsForUser(users.id);
+        if (appList) {
+          let userAppId = getAppId(appList);
           if (userAppId == undefined) {
-            var appInstalled = client.installAppForUser(users.id,appId)
-            if(appInstalled)
-            {
-              axios.post("https://graph.microsoft.com/v1.0/users/" + users.id + "/teamwork/sendActivityNotification", postData, {
-                headers: {
-                  "accept": "application/json",
-                  "contentType": 'application/json',
-                  "authorization": "bearer " + applicationToken
-                }
-              })
-              .then(res => {
-                console.log(`statusCode: ${res.status}`)
-                if(res.status == 429)
-                {
-                  handleTooManyRequestError(users.id, postData);
-                }
-              })
-              .catch(error=>{
-              console.log(error);
-            });
-            }
-          }
-          else{
-            axios.post("https://graph.microsoft.com/v1.0/users/" + users.id + "/teamwork/sendActivityNotification", postData, {
-              headers: {
-                "accept": "application/json",
-                "contentType": 'application/json',
-                "authorization": "bearer " + applicationToken
-              }
-            })
-            .then(res => {
-              console.log(`statusCode: ${res.status}`)
-              if(res.status == 429)
-              {
-                handleTooManyRequestError(users.id,postData);
-              }
-            })
+            await client.installAppForUser(users.id, appId)
           }
+
+          recipientsList.push({
+            "@odata.type": "microsoft.graph.aadUserNotificationRecipient",
+            "userId": users.id
+          });
+        }
+      }
+      catch (ex) {
+        console.error("Installation failed for " + users.displayName);
       }
+    });
+
+    var recipientsChunks = splitIntoChunks(recipientsList, 85);
```

Comment:
What is 85 number ?
We can add some constant /var or add comment.

Reply:
Done